actors/builtin/market/market_actor.go

```diff
@@ -209,8 +224,18 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 
 			newDealIds = append(newDealIds, id)
 		}
-		st.Proposals = proposals.Root()
-		st.DealIDsByParty = dbp.Root()
+		propc, err := proposals.Root()
+		if err != nil {
+			rt.Abortf(exitcode.ErrIllegalState, "failed to flush proposal set: %w", err)
+		}
+		st.Proposals = propc
```

Comment:
I think you could generally do

```
st.Proposals, err = proposals.Root()
if err != nil { ... }
```
because if we're aborting anyway, the state will be rolled back.